```{r}

# Nearest _Gene

# Load the annotation database
library(org.At.tair.db)


# Loading cHIPseeker library
library(ChIPseeker)
# Loading over libaries
library(GenomicRanges)
library(rtracklayer)


# Load peak data from 3000 peaks GEM file
peak_file <- "C:\\Users\\Tanaya\\OneDrive - University of Toronto\\BCB330Y\\3000_bZIP53_bZIP9.narrowPeak"
peaks <- readPeakFile(peak_file)
# This will read your peak data into a GRanges object, which can then be processed using ChIPseeker.


# Import the GFF3 file (Arabidopsis gene annotations)
gff_file <- "C:\\Users\\Tanaya\\OneDrive - University of Toronto\\BCB330Y\\5' UTR_extracted_+-1000.gff"

gff <- import(gff_file)

# Inspect the GFF file to understand its structure
head(gff)
head(seqnames(gff))


# Making sure both of them have the same chr values

# Remove "Chr" prefix by updating the levels of seqnames
seqlevels(gff) <- gsub("Chr", "", seqlevels(gff))

# Check the seqnames to confirm the change
print(seqnames(gff))


# Find the nearest gene for each peak (based on the GFF file)
nearest_genes <- nearest(peaks, gff)
print(nearest_genes)
# write.csv(nearest_genes, "indexes_nearest.txt", row.names = FALSE)

nearest_genes_info <- gff[nearest_genes]
print(nearest_genes_info)


# Now from that information, make a csv file!

# Index starts at 1 i.e. nearest_genes[1] = 18905
# gff[1] = the feature at that index


# Create an empty data frame to store the results
all_data <- data.frame(bZIP1_AGI = character(), bZIP2_AGI = character(), Promoter_Sequence_AGI = character())

for (indices in nearest_genes) {
  object <- gff[indices]
  AGI_of_DNA <- object$Parent[[1]]
  
  # Create a data frame for this iteration
  data <- data.frame(bZIP1_AGI = 'AT3G62420', bZIP2_AGI = 'AT5G24800', Promoter_Sequence_AGI = AGI_of_DNA)
  
  # bZIP2:AT2G18160
  # bZIP9:AT5G24800
  # bZIP53:AT3G62420
  
  # Append the data to the all_data data frame
  all_data <- rbind(all_data, data)
}

# Write the complete data frame to the CSV file at the end
write.csv(all_data, "bZIP53_bZIP9", row.names = FALSE)

break


# Trying with only a file with no +-1000 range

# Load the annotation database
library(org.At.tair.db)


# Loading cHIPseeker library
library(ChIPseeker)
# Loading over libaries
library(GenomicRanges)
library(rtracklayer)


library(TxDb.Athaliana.BioMart.plantsmart51)  
library(org.At.tair.db) 


# Example of creating a GRanges object from a BED file
peaks_2_9 <- rtracklayer::import("C:\\Users\\Tanaya\\OneDrive - University of Toronto\\BCB330Y\\3000_bZIP2_bZIP9.narrowPeak")

# Import the GFF3 file (Arabidopsis gene annotations)
gff_file <- "C://Users//Tanaya//OneDrive - University of Toronto//BCB330Y//5' UTR_extracted.gff"

gff <- import(gff_file)


# Annotating the peaks with the Arabidopsis annotation
peaks_gff <- annotatePeak(peaks_2_9, 
                          TxDb=TxDb.Athaliana.BioMart.plantsmart51,
                                tssRegion=c(-1000, 1000),  # Promoter region +/- 1000 base pairs
                                annoDb="org.At.tair.db")  # TAIR database for gene annotations

head(peaks_gff)

write.csv(as.data.frame(peaks_gff), "JUST_5'_UTR.csv", row.names = FALSE)


break



# Trying with 5'UTR only but with a gff file
BiocManager::install("txdbmaker")

# Load necessary libraries
library(ChIPseeker)
library(GenomicRanges)
library(rtracklayer)
library(GenomicFeatures)

# Import peak data from the BED file
peaks_2_9 <- rtracklayer::import("C:\\Users\\Tanaya\\OneDrive - University of Toronto\\BCB330Y\\3000_bZIP2_bZIP9.narrowPeak")

peaks_2_9

# Import the GFF3 file and create a TxDb object
gff_file <- "C:\\Users\\Tanaya\\OneDrive - University of Toronto\\BCB330Y\\TAIR10_GFF3_genes.gff"
txdb <- txdbmaker::makeTxDbFromGFF(file = gff_file, format = "gff")
transcripts(txdb)
exons(txdb)


# Annotate peaks using the TxDb object
peaks_gff <- annotatePeak(peaks_2_9, TxDb = txdb, tssRegion = c(-1000, 1000),  annoDb="org.At.tair.db")

# Save the annotated peaks as a CSV file
write.csv(as.data.frame(peaks_gff), "5'_UTR_with_gff.csv", row.names = FALSE)



## Trying something else

# Load necessary libraries
library(ChIPseeker)
library(GenomicRanges)
library(rtracklayer)
library(GenomicFeatures)


# Example of creating a GRanges object from a BED file
peaks_2_9 <- rtracklayer::import("C:\\Users\\Tanaya\\OneDrive - University of Toronto\\BCB330Y\\3000_bZIP2_bZIP9.narrowPeak")

# Import the GFF3 file (Arabidopsis gene annotations)
gff_file <- "C://Users//Tanaya//OneDrive - University of Toronto//BCB330Y//5' UTR_extracted_+-1000.gff"

gff <- import(gff_file)


# Annotating the peaks with the Arabidopsis annotation
peaks_gff <- annotatePeak(peaks_2_9, gff,
                                tssRegion=c(-1000, 1000),  # Promoter region +/- 1000 base pairs
                                annoDb="org.At.tair.db")  # TAIR database for gene annotations

write.csv(as.data.frame(peaks_gff), "TRIAL", row.names = FALSE)











```


